libs/template/experimental/virtual-scrolling/src/lib/model.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  ChangeDetectorRef,
+  Directive,
+  EmbeddedViewRef,
+  ErrorHandler,
+  NgIterable,
+  NgZone,
+  Output,
+  TemplateRef,
+  TrackByFunction,
+  ViewContainerRef,
+} from '@angular/core';
+import { RxStrategies } from '@rx-angular/cdk/render-strategies';
+import { RxDefaultListViewContext } from '@rx-angular/cdk/template';
+import { Observable, Subject } from 'rxjs';
+
+type CreateViewContext<T, C, U> = (value: T, computedContext: U) => C;
+
+type UpdateViewContext<T, C, U> = (
+  value: T,
+  view: EmbeddedViewRef<C>,
+  computedContext?: U
+) => void;
+
+export interface TemplateSettings<T, C, U> {
+  viewContainerRef: ViewContainerRef;
+  templateRef: TemplateRef<C>;
+  createViewContext: CreateViewContext<T, C, U>;
+  updateViewContext: UpdateViewContext<T, C, U>;
+  viewCacheSize: number;
+}
+
+export interface RenderSettings {
+  cdRef: ChangeDetectorRef;
+  parent: boolean;
+  patchZone?: NgZone;
+  strategies: RxStrategies<string>;
+  defaultStrategyName: string;
+  errorHandler?: ErrorHandler;
+}
+
+export const enum ListTemplateChangeType {
+  insert,
+  remove,
+  move,
+  update,
+  context,
```

Comment:
Would it be better to use a union type of strings to save some bytes?

Reply:
same here, I want to get rid of it